beetsplug/convert.py

```diff
@@ -187,31 +187,13 @@ def auto_convert(self, config, task):
 
     def auto_convert_keep(self, config, task):
         if self.config['auto_keep']:
-            fmt = self.config['format'].as_str().lower()
-
-            dest = self.config['dest'].get()
-            if not dest:
-                raise ui.UserError('no convert destination set')
-            dest = util.bytestring_path(dest)
-
-            path_formats = ui.get_path_formats(self.config['paths'] or None)
-
-            hardlink = self.config['hardlink'].get(bool)
-            link = self.config['link'].get(bool)
-
-            threads = self.config['threads'].get(int)
+            empty_opts = self.commands()[0].parser.get_default_values()
```

Comment:
Is there a better way to do this? It doesn't feel quite right.

Reply:
Very creative! While it doesn't bother me too much, it does seem a little odd that we'd be using the CLI arguments here even though this method doesn't have anything to do with the command itself.

It's not exactly very clean, but one option would be to let `_get_opts_and_config`'s argument be `None`. I think all of the conditionals in there would end up ignoring the `opts` value for this mode anyway, right? At least it would make it clear that we are using only the configuration values, nothing from the command line.

Reply:
Actually that was my first attempt but it doesn't work.

For example for dest it will do:
`dest = None.dest ....`
Resulting in: `'NoneType' object has no attribute 'dest'`

A elegant workaround would have been to to this: `dest = opts?.dest` for every options. But unfortunately safe navigation operator is not available in python.

The only option left then would be to manually check if opts is None for every option witch is not very elegant in my opinion. What do you think? I think `parser.get_default_values()` works well in this context but looks very hacky at the same time.

Reply:
Ah, yes, the `opts.foo if opts else config[‘foo’].get()` dance was pretty much what I was recommending here. But you make a good point that this would bring a high complexity tax for something that doesn’t seem worth it.

I think a better solution would probably be to [use Confuse’s support for argparse options](https://confuse.readthedocs.io/en/latest/usage.html#command-line-options). But maybe the best route would be to note that this is the next step and merge this PR as-is, hoping to tackle the Confuse-based alternative in a subsequent PR (since it will involve some refactoring of its own)?

Reply:
Fine with me